test/tests_for_integration/api_to_db_test.py

```diff
@@ -1324,8 +1338,9 @@ def test_run_job_batch(ee2_port, ws_controller, mongo_client):
             parent_job_id=parent_job_id,
             app_id=None,
             app_module=None,
+            wsid=job_batch_wsid,
         )
-        expected_sub_1["+KB_WSID"] = ""
+        expected_sub_1["+KB_WSID"] = f'"{job_batch_wsid}"'
```

Comment:
Do we have a test case for a batch job without a workspace ID anymore?

Reply:
I don't think we ever had a test case for that, as job_batch_params always specifies a wsid in each of these tests, so it should always have a workspace id. I can write a test, but currently having an issue with adding another jobrun

no_wsid_ret = ee2.run_job_batch([job1_params, job2_params], job_batch_params_no_wsid)

results in

```
>                   raise ServerError(**err["error"])
E                   installed_clients.baseclient.ServerError: Server error: -32000. ()
E                   Traceback (most recent call last):
E                     File "/ee2/lib/execution_engine2/execution_engine2Server.py", line 107, in _call_method
E                       result = method(ctx, *params)
E                     File "/ee2/lib/execution_engine2/execution_engine2Impl.py", line 367, in run_job_batch
E                       job_ids = mr.run_job_batch(
E                     File "/ee2/lib/execution_engine2/sdk/SDKMethodRunner.py", line 328, in run_job_batch
E                       return self.get_runjob().run_batch(
E                     File "/ee2/lib/execution_engine2/sdk/EE2Runjob.py", line 381, in run_batch
E                       self._add_job_requirements(params, bool(as_admin))  # as_admin checked above
E                     File "/ee2/lib/execution_engine2/sdk/EE2Runjob.py", line 421, in _add_job_requirements
E                       job[_JOB_REQUIREMENTS] = jrr.resolve_requirements(
E                     File "/ee2/lib/execution_engine2/utils/job_requirements_resolver.py", line 412, in resolve_requirements
E                       cat_reqs_all = self._get_catalog_reqs(module_name, function_name, catalog_cache)
E                     File "/ee2/lib/execution_engine2/utils/job_requirements_resolver.py", line 469, in _get_catalog_reqs
E                       group_config = catalog_cache.lookup_job_resource_requirements(
E                     File "/ee2/lib/execution_engine2/utils/catalog_cache.py", line 91, in lookup_job_resource_requirements
E                       ] = self.get_catalog().list_client_group_configs(
E                     File "<string>", line 3, in list_client_group_configs
E                     File "/miniconda-latest/lib/python3.8/unittest/mock.py", line 1081, in __call__
E                       return self._mock_call(*args, **kwargs)
E                     File "/miniconda-latest/lib/python3.8/unittest/mock.py", line 1085, in _mock_call
E                       return self._execute_mock_call(*args, **kwargs)
E                     File "/miniconda-latest/lib/python3.8/unittest/mock.py", line 1142, in _execute_mock_call
E                       result = next(effect)
E                     File "/miniconda-latest/lib/python3.8/unittest/mock.py", line 388, in __next__
E                       return next(self.obj)
E                   StopIteration
```
Which I am trying to figure out... Does it have something to do with the mock side effects not returning enough values?

Reply:
> I don't think we ever had a test case for that

Looks like you're right

Reply:
Are you doing that in the same test? If so, and the test mocks have side effects, then yeah, that's what's happening. I (almost) always only run 1 method per test, and factor out any common setup / result or mock checking code into helper methods.

Reply:
Yes. Is it possible to factor out any common setup with these mocks? Is it possible to move these patching set code into another function? Or should I keep adding things to the side_effects. Or do you want me to copy and paste this test and then just change the part with the wsid?

Reply:
> Is it possible to factor out any common setup with these mocks? Is it possible to move these patching set code into another function? 

Should be, I did it with the non-batch functions (and I think some of the batch functions as well. 

> Or should I keep adding things to the side_effects.

I personally avoid that like the plague - the tests get too hard to understand if you're splitting the side effects between multiple calls.

Reply:
Although another approach might be to C&P and then factor out common stuff in another PR